### PR TITLE
models: show default LLM first in dropdown

### PR DESCRIPTION
**Pull request by keegancsmith**
The order we list our models in dotcom.ts is the order it is shown in. I imagine in future we likely want to order these such that LLMs we want to guide users to use are higher up. For now I just moved the default LLM to the top.

Alternatively we should consider making the selected LLM at the top of the dropdown list?

Test Plan: CI

Fixes https://github.com/sourcegraph/cody/issues/3820

### CODE REVIEW

**Comment by keegancsmith** [comment]
Hey I'd love to pair to see how I can manually test this :)

**Comment by philipp-spiess** [comment]
@keegancsmith happy to chat :)

**Review by philipp-spiess — APPROVED** [review]
LGTM but we need to recreate agent snapshots here:

```
source agent/scripts/export-cody-http-recording-tokens.sh
pnpm update-agent-recordings
```

**Comment by philipp-spiess** [comment]
> Alternatively we should consider making the selected LLM at the top of the dropdown list?

I think for now this is fine, we plan some larger scale updates to the input soon: https://github.com/sourcegraph/cody/issues/3792

**Comment by keegancsmith** [comment]
@philipp-spiess this made bigger changes than I expected, but I suppose we had tests that just used the first model hence the need to update?

**Comment by philipp-spiess** [comment]
@keegancsmith Yeah it seems so. This seems reasonable though, one less Claude 2 call to worry about 🧠

**Comment by philipp-spiess** [comment]
Looks like the E2E tests also depend on this 😬 If you want to fix them, I recommend running them with `--debug` locally so that you can see which selector breaks. Happy to take over from here as well if you want.

```bash
cd vscode
pnpm test:e2e --debug -- attribution
```

**Comment by keegancsmith** [comment]
> Happy to take over from here as well if you want.

You can start worrying, I plan on contributing much more to this repo for the next while :) So I'll fix.

File: lib/shared/src/models/dotcom.ts [hunk 1]
```diff
@@ -10,6 +10,16 @@ import { ModelUsage } from './types'
...
+    // The order listed here is the order shown to users. Put the default LLM first.
+    {
+        title: 'Claude 3 Sonnet',
+        model: 'anthropic/claude-3-sonnet-20240229',
+        provider: 'Anthropic',
+        default: true,
+        codyProOnly: false,
+        usage: [ModelUsage.Chat, ModelUsage.Edit],
+        contextWindow: { input: CHAT_INPUT_TOKEN_BUDGET, output: CHAT_OUTPUT_TOKEN_BUDGET },
+    },
```
Comment (abeatrix) [inline lib/shared/src/models/dotcom.ts:22]: If we want to order the list based on preferences, should we move all Claude 3 models to the top too?

**Review by abeatrix — APPROVED** [review]
Would it be better to sort the [model list inside the UI component ](https://sourcegraph.com/github.com/sourcegraph/cody/-/blob/vscode/webviews/Components/ChatModelDropdownMenu.tsx?L95)instead since the default model could change based on user's last picked model? 🤔

**Comment by philipp-spiess** [comment]
> Would it be better to sort the model list inside the UI component instead since the default model could change based on user's last picked model? 

Then it wouldn't work in JetBrains though, right? I don't think it's expected in a dropdown that your currently selected option is always up front. I think the first slot should be _our recommendation_. WDYT?

**Comment by keegancsmith** [comment]
@philipp-spiess please take a look, the issue seemed to be in the naive way the mock-server decided which message was from the human. I am guessing how we format the message to the mock-server changed when I put opus as the first model.

**Review by abeatrix — APPROVED** [review]
> Then it wouldn't work in JetBrains though, right? I
@philipp-spiess I think they are already sorting it in their UI based on the latest version of the JB plugin 🤔 
image

@keegancsmith Thanks for updating the list! I didn't know changing the model list would require that many changes 😱 The list aligns with what JB has so LGTM!

**Pull request merged**